src/execution/riscv64/simulator-riscv64.cc

```diff
@@ -2837,22 +2837,22 @@ void Simulator::DecodeRVIType() {
     case RO_LBU: {
       int64_t addr = rs1() + imm12();
       uint8_t val = ReadMem<uint8_t>(addr, instr_.instr());
-      set_rd(zext_xlen(val), false);
+      set_rd(val, false);
```

Comment:
The commit message says "Load a unsinged num should not sign-extend.", but `zext_xlen` is zero-extending. Also, for the 64-bit simulator, this is essentially doing nothing except casting to a `uint64_t`:

```cpp
using reg_t = uint64_t;
#define xlen 64
...
#define zext_xlen(x) (((reg_t)(x) << (64 - xlen)) >> (64 - xlen))
```

Does this change actually affect anything?

Reply:
As Brice said, `zext_xlen(..)` means to zero-extend to `xlen` as opposed to `sext_xlen(...)` for sign-extension. So all the fixes on `zext_xlen` is unnecessary.